### PR TITLE
updated contract to 1.4.1 and rust 1.57, added locked contract

### PR DESCRIPTION
**Pull request by deuszex**
Updated casper and rust version.
Added a new contract to demonstrate locking contracts.
Added some details about upgrading contract.

### CODE REVIEW

**Comment by cspramit** [comment]
@ramujin pls review

**Comment by ramujin** [comment]
I am not sure why this was assigned to me as it is @deuszex's PR request. I'm moving myself to reviewer and placing him as the assignee.

@deuszex I noticed that you have hardcoded a Linux dependency in the Makefile that breaks the code for me on my Mac. Line 2 of the Makefile reads `rustup default nightly-2021-09-17-x86_64-unknown-linux-gnu`. If you want to ensure the nightly build is updated, you can issue commands similar to what we have in the Counter docs: https://casper.network/docs/dapp-dev-guide/tutorials/counter/setup.

I'm happy to test and review once it's ready :)

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,12 @@
 prepare:
+	rustup default nightly-2021-09-17-x86_64-unknown-linux-gnu
```
Comment (ramujin) [inline Makefile:2]: This line breaks Rust the installation on a Mac. After running the make commands, I had to reinstall Rust from scratch in order to get NCTL to work again.

**Review by ramujin — CHANGES_REQUESTED** [review]

**Comment by devendran-m** [comment]
Closing as won't do

**Pull request closed without merging**